src/Computation/Values/And.cs

```diff
@@ -22,7 +22,9 @@ public override BitVecExpr AsBitVector(Context context)
 
     public override BoolExpr AsBool(Context context)
     {
-        return context.MkAnd(_left.AsBool(context), _right.AsBool(context));
+        return _left is Bool || _right is Bool
+            ? context.MkAnd(_left.AsBool(context), _right.AsBool(context))
```

Comment:
Does this to unnecessary constraints being generated for the one that is already a `Bool`?

Reply:
Yeah, I was meaning that if one side is already `Bool` then can we avoid calling `AsBool` on it, but I see that as `AsBool` converts to `BoolExpr` so it's needed regardless.